File: backend/src/main/java/com/quickstay/saga/dto/SagaBookingRequest.java
package com.quickstay.saga.dto;

import com.quickstay.booking.dto.ReservationRequest;
import jakarta.validation.Valid;
import jakarta.validation.constraints.DecimalMin;
import jakarta.validation.constraints.NotNull;

import java.math.BigDecimal;

public record SagaBookingRequest(
        @Valid @NotNull ReservationRequest reservation,
        @NotNull @DecimalMin("0.01") BigDecimal paymentAmount,
        boolean failPayment
) {}

File: quickstay_mobile/lib/core/models/saga_booking_request.dart
/// Espeja com.quickstay.saga.dto.SagaBookingRequest, que a su vez envuelve
/// com.quickstay.booking.dto.ReservationRequest.
class SagaBookingRequest {
  final String roomId;
  final String guestFullName;
  final String guestEmail;
  final DateTime checkIn;
  final DateTime checkOut;
  final double paymentAmount;
  final bool failPayment;

  SagaBookingRequest({
    required this.roomId,
    required this.guestFullName,
    required this.guestEmail,
    required this.checkIn,
    required this.checkOut,
    required this.paymentAmount,
    this.failPayment = false,
  });

  static String _fmtDate(DateTime d) {
    final mm = d.month.toString().padLeft(2, '0');
    final dd = d.day.toString().padLeft(2, '0');
    return '${d.year}-$mm-$dd';
  }

  Map<String, dynamic> toJson() {
    return {
      'reservation': {
        'roomId': roomId,
        'guestFullName': guestFullName,
        'guestEmail': guestEmail,
        'checkIn': _fmtDate(checkIn),
        'checkOut': _fmtDate(checkOut),
      },
      'paymentAmount': paymentAmount,
      'failPayment': failPayment,
    };
  }
}
